acme/acme/crypto_util.py

```diff
@@ -237,10 +247,15 @@ def make_csr(private_key_pem: bytes, domains: Optional[Union[Set[str], List[str]
     params ordered this way for backward competablity when called by positional argument.
     :returns: buffer PEM-encoded Certificate Signing Request.
     """
-    private_key = crypto.load_privatekey(
-        crypto.FILETYPE_PEM, private_key_pem)
-    csr = crypto.X509Req()
-    sanlist = []
+    # builder.sign(...) below will error out if this isn't the correct private key type
+    private_key: CertificateIssuerPrivateKeyTypes = cast(
```

Comment:
yeah, it's mypy yelling:

```
acme/acme/crypto_util.py:274: error: Argument 1 to "sign" of "CertificateSigningRequestBuilder" has incompatible type "DHPrivateKey | Ed25519PrivateKey | Ed448PrivateKey | RSAPrivateKey | DSAPrivateKey | EllipticCurvePrivateKey | X25519PrivateKey | X448PrivateKey"; expected "Ed25519PrivateKey | Ed448PrivateKey | RSAPrivateKey | DSAPrivateKey | EllipticCurvePrivateKey"  [arg-type]
```

this seems to be due to a mypy bug where it can't check against union types -- in fact, someone in the [issue](https://github.com/python/mypy/issues/12155#issuecomment-2074271526) seems to be running into this w/ cryptography as well. looks like the latest mypy version has a fix, so i'll try upgrading it

Reply:
nevermind, it's still broken on the latest mypy (https://github.com/python/mypy/pull/17371#issuecomment-2243200563). i guess i'll just throw in a mypy ignore here?